flocker/node/_deploy.py

```diff
@@ -143,6 +146,16 @@ class StartApplication(object):
     :ivar NodeState node_state: The state of the node the ``Application``
         is running on.
     """
+    application = field(type=Application, mandatory=True)
+    node_state = field(type=NodeState, mandatory=True)
+
+    @property
```

Comment:
Ick. An attribute that is *required* to be side-effect-y isn't great. I suggest `IStateChange.eliot_action` be a method, not an attribute.